### PR TITLE
Migration to python social auth #141

### PR DESCRIPTION
**Pull request by mapeveri**
Migration from django-social-auth to python-social-auth. #141

### CODE REVIEW

**Comment by nemesisdesign** [comment]
Looking good! It only remains to update a few tests that still try to do ´reverse´ lookup for  ´social_auth´ urls.

File: nodeshot/ui/default/templates/base.html [hunk 1]
```diff
@@ -127,38 +127,30 @@
                 {% blocktrans %}By signing up you declare to have accepted our <a href="{{ PRIVACY_POLICY_LINK }}" class="js-dismiss">privacy&nbsp;policy</a><br />and <a href="{{ TERMS_OF_SERVICE_LINK }}" class="js-dismiss">terms&nbsp;of&nbsp;service</a>.{% endblocktrans %}
             </p>
 
-            {% if SOCIAL_AUTH_ENABLED %}
```
Comment (nemesisdesign) [inline nodeshot/ui/default/templates/base.html]: remember to put this back

File: nodeshot/ui/default/templates/base.html [hunk 2]
```diff
@@ -127,38 +127,30 @@
...
             <p class="social-buttons">
-                {% if FACEBOOK_ENABLED %}
```
Comment (nemesisdesign) [inline nodeshot/ui/default/templates/base.html]: remember to put this back

File: nodeshot/ui/default/templates/base.html [hunk 3]
```diff
@@ -127,38 +127,30 @@
...
-                <a href="{% url "socialauth_begin" "facebook" %}" class="btn btn-social btn-facebook">
+                <a href="{% url 'social:begin' 'facebook' %}" class="btn btn-social btn-facebook">
                     <span class="icon icon-facebook"></span>
                     <span class="text">{% trans 'Sign up with Facebook' %}</span>
                     <span class="hover">&nbsp;</span>
                 </a>
-                {% endif %}
 
-                {% if GOOGLE_ENABLED %}
```
Comment (nemesisdesign) [inline nodeshot/ui/default/templates/base.html]: remember to put this back

File: nodeshot/ui/default/templates/base.html [hunk 4]
```diff
@@ -127,38 +127,30 @@
...
-                <a href="{% url "socialauth_begin" "google-oauth2" %}" class="btn btn-social btn-google">
+                <a href="{% url 'social:begin' 'google-oauth2' %}" class="btn btn-social btn-google">
                     <span class="icon icon-google"></span>
                     <span class="text">{% trans 'Sign up with Google' %}</span>
                     <span class="hover">&nbsp;</span>
                 </a>
-                {% endif %}
 
-                {% if GITHUB_ENABLED %}
```
Comment (nemesisdesign) [inline nodeshot/ui/default/templates/base.html]: remember to put this back

File: nodeshot/ui/default/templates/base.html [hunk 5]
```diff
@@ -204,38 +196,30 @@
          aria-hidden="true">
         <div class="modal-dialog modal-content modal-body inverse user-dialog">
 
-            {% if SOCIAL_AUTH_ENABLED %}
```
Comment (nemesisdesign) [inline nodeshot/ui/default/templates/base.html]: remember to put this back

File: nodeshot/ui/default/templates/base.html [hunk 6]
```diff
@@ -204,38 +196,30 @@
...
             <p class="social-buttons">
-                {% if FACEBOOK_ENABLED %}
```
Comment (nemesisdesign) [inline nodeshot/ui/default/templates/base.html]: remember to put this back

File: nodeshot/ui/default/templates/base.html [hunk 7]
```diff
@@ -204,38 +196,30 @@
...
-                <a href="{% url "socialauth_begin" "facebook" %}" class="btn btn-social btn-facebook">
+                <a href="{% url 'social:begin' 'facebook' %}" class="btn btn-social btn-facebook">
                     <span class="icon icon-facebook"></span>
                     <span class="text">{% trans 'Sign in with Facebook' %}</span>
                     <span class="hover">&nbsp;</span>
                 </a>
-                {% endif %}
 
-                {% if GOOGLE_ENABLED %}
```
Comment (nemesisdesign) [inline nodeshot/ui/default/templates/base.html]: remember to put this back

File: nodeshot/ui/default/templates/base.html [hunk 8]
```diff
@@ -204,38 +196,30 @@
...
-                <a href="{% url "socialauth_begin" "google-oauth2" %}" class="btn btn-social btn-google">
+                <a href="{% url 'social:begin' 'google-oauth2' %}" class="btn btn-social btn-google">
                     <span class="icon icon-google"></span>
                     <span class="text">{% trans 'Sign in with Google' %}</span>
                     <span class="hover">&nbsp;</span>
                 </a>
-                {% endif %}
 
-                {% if GITHUB_ENABLED %}
```
Comment (nemesisdesign) [inline nodeshot/ui/default/templates/base.html]: remember to put this back

File: nodeshot/conf/settings.py [hunk 9]
```diff
@@ -333,44 +339,51 @@
...
+    SOCIAL_AUTH_FORCE_EMAIL_VALIDATION = True
+
+    SOCIAL_AUTH_LOGIN_URL = '/'
+    SOCIAL_AUTH_LOGIN_REDIRECT_URL = '/'
+    SOCIAL_AUTH_LOGIN_ERROR_URL = '/'
 
-    LOGIN_URL = '/'
-    LOGIN_REDIRECT_URL = '/'
-    LOGIN_ERROR_URL = '/'
+    SOCIAL_AUTH_PIPELINE = (
```
Comment (nemesisdesign) [inline nodeshot/conf/settings.py]: could you move this up after `AUTHENTICATION_BACKENDS`porfavor?

**Comment by nemesisdesign** [comment]
check this file too:
https://github.com/ninuxorg/nodeshot/blob/master/nodeshot/ui/default/settings.py#L64

File: nodeshot/conf/settings.py (hunk 9 above)
Comment (mapeveri) [inline nodeshot/conf/settings.py]: I do not understand. What do you mean?

Comment (nemesisdesign) [inline nodeshot/conf/settings.py]: solo mover el block `SOCIAL_AUTH_PIPELINE` justo delante de `AUTHENTICATION_BACKENDS`

**Pull request merged**